redisson/src/main/java/org/redisson/RedissonBaseLock.java

```diff
@@ -174,7 +174,13 @@ protected void scheduleExpirationRenewal(long threadId) {
             oldEntry.addThreadId(threadId);
         } else {
             entry.addThreadId(threadId);
-            renewExpiration();
+            try {
+                renewExpiration();
+            } finally {
+                if (Thread.currentThread().isInterrupted()) {
```

Comment:
If the thread has been interrupted at this point, it cancels the expiration renewal